src/adapters/Cornerstone/Segmentation_4X.js

```diff
@@ -1105,10 +1103,7 @@ function insertPixelDataPlanar(
             continue;
         }
 
-        const sourceImageMetadata = cornerstone.metaData.get(
-            "instance",
-            imageId
-        );
+        const sourceImageMetadata = metadataProvider.get("instance", imageId);
```

Comment:
Thanks @Punzo for applying the changes so fast! I have doubt, though. Who is adding support for the "instance" type?

Reply:
> Thanks @Punzo for applying the changes so fast! 

you are welcome,

> I have doubt, though. Who is adding support for the "instance" type?

 not sure I have understood what do you mean. the cornerstone provider does provide all the metadata of the instance when using metadataProvider.get("instance", imageId);

Reply:
When trying it out, `sourceImageMetadata` is undefined. However, if I comment out the ROWS-COLUMNS check, the segmentation loads correctly. 

I debugged the `metadaProvider.get` method, and it seems there's no registered provider that responds to the `instance` type. Thus, I was wondering if I miss any initialization that adds a provider to have the instance type available. 

Thanks

Reply:
Hey @Punzo, I would appreciate if you could help me on this. Thanks!

Reply:
Hi, 

you can check 

https://github.com/OHIF/Viewers/blob/master/extensions/cornerstone/src/init.js for cornerstone initialization.

here is where the segmentation is parsed in OHIF using dcmjs:
https://github.com/OHIF/Viewers/blob/c554c5fac2dfb8cf4800ceffd936475e07ca7782/extensions/dicom-segmentation/src/getOHIFDicomSegSopClassHandler.js#L152

and 

Can you also move thiss discussion to an issue in the issue tracker, please?